### PR TITLE
Meme API 

### PR DESCRIPTION
**Pull request by xiningchen**
Completed
- Creates memes
- Upvote/downvote meme vote
- deletes meme's vote
- retrieves details about a meme
---
Remaining:
- retrieves a list of memes sorted by popularity?

### CODE REVIEW

File: routes/api/memes/index.js [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+       const msg = (err && err.errors && err.errors[0] && err.errors[0].message) || 'Failed to create meme';
+       res.status(400).json({error: msg});
+   });
+});
+
+/**
+ * Gets lists of memes
+ */
+router.get('/', async (req, res) => {
+    // const sort = (['top', 'host', 'new'].includes(req.query.sort) && req.query.sort) || 'top';
```
Comment (Step7750) [inline routes/api/memes/index.js]: Please remove these comments

File: routes/api/memes/index.js [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+/**
+ * Creates meme
+ */
+router.post('/', auth.isAuthenticated, (req, res) => {
+   models.Meme.create({
+        title: req.body.title,
+        link: req.body.link,
+        creatorId: req.session.userId,
+        TemplateId: parseInt(req.body.templateId) || null,
+        CommunityId: parseInt(req.body.communityId) || null,
```
Comment (Step7750) [inline routes/api/memes/index.js]: Could you please add an explicit check to ensure that the passed in communityId exists and is an int? Sequelize apparently allows nulls here when we should force a meme to be part of a community.

**Comment by xiningchen** [comment]
Should I wait to merge? There's still the get list of meme functionality left...

**Comment by Step7750** [comment]
I think this is a good start, the other meme functionality will need a partial refactor of the meme voting so I think this should be merged first since it'll take some time for that.

**Comment by xiningchen** [comment]
Ok I just set up the method then... It's pretty much empty. If you're ok with it, I'll merge.

**Pull request merged**